core/chains/evm/txmgr/confirmer_test.go

```diff
@@ -193,1249 +183,327 @@ func TestEthConfirmer_Lifecycle(t *testing.T) {
-		})).Return(nil).Run(func(args mock.Arguments) {
-			elems := args.Get(1).([]rpc.BatchElem)
-			*(elems[0].Result.(*evmtypes.Receipt)) = txmReceipt
-			elems[0].Error = errors.New("foo")
-		}).Once()
+	t.Run("marks re-org'd terminally stuck transaction as unconfirmed, marks latest attempt as in-progress, deletes receipt, removed error", func(t *testing.T) {
+		ethKeyStore := cltest.NewKeyStore(t, db).Eth()
+		_, fromAddress := cltest.MustInsertRandomKeyReturningState(t, ethKeyStore)
+		// Insert terminally stuck transaction that stays fatal error
+		etx := mustInsertTerminallyStuckTxWithAttempt(t, txStore, fromAddress, 0, blockNum)
+		mustInsertEthReceipt(t, txStore, blockNum, utils.NewHash(), etx.TxAttempts[0].Hash)
+		ec := newEthConfirmer(t, txStore, ethClient, cfg, evmcfg, ethKeyStore, nil)
 
-		// No error because it is merely logged
-		require.NoError(t, ec.CheckForReceipts(ctx, blockNum, latestFinalizedBlockNum))
+		ethClient.On("SequenceAt", mock.Anything, fromAddress, mock.Anything).Return(evmtypes.Nonce(0), nil).Maybe()
+		require.NoError(t, ec.CheckForConfirmation(ctx, &head))
 
-		etx, err := txStore.FindTxWithAttempts(ctx, etx1.ID)
+		var err error
+		etx, err = txStore.FindTxWithAttempts(ctx, etx.ID)
 		require.NoError(t, err)
-		require.Len(t, etx.TxAttempts, 1)
-		require.Len(t, etx.TxAttempts[0].Receipts, 0)
+		require.Equal(t, txmgrcommon.TxUnconfirmed, etx.State)
+		require.Equal(t, "", etx.Error.String)
+		attempt := etx.TxAttempts[0]
+		require.Equal(t, txmgrtypes.TxAttemptInProgress, attempt.State)
+		require.Equal(t, 0, len(attempt.Receipts))
 	})
 
-	etx2 := cltest.MustInsertUnconfirmedEthTxWithBroadcastLegacyAttempt(t, txStore, nonce, fromAddress)
-	nonce++
-	require.Len(t, etx2.TxAttempts, 1)
-	attempt2_1 := etx2.TxAttempts[0]
-	require.Len(t, attempt2_1.Receipts, 0)
-
-	t.Run("saves eth_receipt and marks eth_tx as confirmed when geth client returns valid receipt", func(t *testing.T) {
-		txmReceipt := evmtypes.Receipt{
-			TxHash:           attempt1_1.Hash,
-			BlockHash:        testutils.NewHash(),
-			BlockNumber:      big.NewInt(42),
-			TransactionIndex: uint(1),
-			Status:           uint64(1),
-		}
-
-		ethClient.On("SequenceAt", mock.Anything, mock.Anything, mock.Anything).Return(evmtypes.Nonce(10), nil)
-		ethClient.On("BatchCallContext", mock.Anything, mock.MatchedBy(func(b []rpc.BatchElem) bool {
-			return len(b) == 2 &&
-				cltest.BatchElemMatchesParams(b[0], attempt1_1.Hash, "eth_getTransactionReceipt") &&
-				cltest.BatchElemMatchesParams(b[1], attempt2_1.Hash, "eth_getTransactionReceipt")
-		})).Return(nil).Run(func(args mock.Arguments) {
-			elems := args.Get(1).([]rpc.BatchElem)
-			// First transaction confirmed
-			*(elems[0].Result.(*evmtypes.Receipt)) = txmReceipt
-			// Second transaction still unconfirmed
-			elems[1].Result = &evmtypes.Receipt{}
-		}).Once()
+	t.Run("handles multiple transactions at a time", func(t *testing.T) {
```

Comment:
There is another test case below with the same name, but slightly different behaviour. Perhaps rename one of the two.